libs/contrib/Data/String/BuildExpressionParser.idr

```diff
@@ -0,0 +1,132 @@
+-- Port of https://hackage.haskell.org/package/parsec-3.1.13.0/docs/Text-Parsec-Expr.html
+-- Original License	BSD-style: https://hackage.haskell.org/package/parsec-3.1.13.0/src/LICENSE
+
+module Data.String.BuildExpressionParser
+
+import Control.Monad.Identity
+import Data.String.Parser
+
+fail : Monad m => String -> ParseT m a
+fail x = P $ \s => do pure $ Fail s.pos x
+
+public export
+data Assoc = AssocNone
+           | AssocLeft
+           | AssocRight
+
+public export
+data Operator a = Infix (Parser (a -> a -> a)) Assoc
+                | Prefix (Parser (a -> a))
+                | Postfix (Parser (a -> a))
+
+public export
+LO : Type -> Type
+LO a = List (Operator a)
```

Comment:
That's a cryptic name, it would be clearer to just use `List (Operator a)`.

Reply:
Updated.